core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -702,11 +762,12 @@ private[storage] class PartiallySerializedBlock[T](
    */
   def finishWritingToStream(os: OutputStream): Unit = {
     ByteStreams.copy(unrolled.toInputStream(), os)
```

Comment:
There's actually a bug here: we aren't guaranteed to free the actual offHeap memory here. I think that in my benchmarks it just so happened to be cleaned by the `sun.misc.Cleaner` that I attached to the DirectByteBuffer in my custom allocator. I'm going to take a closer look at this block to try to fix the leak issues here.